### PR TITLE
feat(sentry): publish all error to sentry

### PR DESCRIPTION
**Pull request by zulkhair**
Closes: WORLD-717

## Overview
  
Integrating world-cli with sentry to track what error happened in the user environment

## Brief Changelog
  
- Create function `logger.ErrorWithSentry` that wrap logger error and sentry capture exception
- Wrap all return err line with `logger.ErrorWithSentry` function
- Init sentry in the main.go file

## Testing and Verifying

Tested sample Error and Panic

## Summary by CodeRabbit

- **New Features**
  - Implemented a new custom logging system with enhanced error reporting.

- **Bug Fixes**
  - Improved error handling across different modules for consistency and better tracking.

- **Refactor**
  - Transitioned from `zerolog` to a custom logger package throughout the application.

- **Documentation**
  - Updated comments to clarify the purpose of `DefaultCallerSkipFrameCount` setting.

- **Chores**
  - Integrated Sentry for real-time error tracking and reporting.
  - Ensured proper flushing of error events before application shutdown.

### CODE REVIEW

**Comment by zulkhair** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #24** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Argus-Labs/world-cli/24?utm_source=stack-comment).

File: cmd/world/main.go [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
 
 func main() {
+	// Sentry initialization
+	err := sentry.Init(sentry.ClientOptions{
+		Dsn:                "{input your DSN here}",
+		EnableTracing:      true,
+		TracesSampleRate:   1.0,
+		ProfilesSampleRate: 1.0,
+		AttachStacktrace:   true,
+	})
```
[the remark this answers is not in the document]
Comment (zulkhair) [reply cmd/world/main.go:20]: it will be put when building apps

File: cmd/world/main.go [hunk 2]
```diff
@@ -1,9 +1,40 @@
...
+		err := recover()
+
+		if err != nil {
+			sentry.CurrentHub().Recover(err)
+
+			// Flush buffered events before the program terminates.
+			// Set the timeout to the maximum duration the program can afford to wait.
+			sentry.Flush(time.Second * 5)
+		}
+	}()
```
[the remark this answers is not in the document]
Comment (zulkhair) [reply cmd/world/main.go:37]: it will not be configurable

**Review by smsunarto — COMMENTED** [review]
It's better to just bubble up the error normally to main.go/root.go because it's easy to forget/tiring to have to use logger.ErrorWithSentry all the time. 

You can then just use the zerolog hooks (https://github.com/rs/zerolog?tab=readme-ov-file#hooks) to inject Sentry there

**Comment by zulkhair** [comment]
I once tried just sending on main.go but only the main.go file as the caller was sent as a stack trace to sentry.

I will try zerolog hooks, i think it will solved the problem.

File: cmd/world/main.go [hunk 3]
```diff
@@ -1,9 +1,40 @@
...
+
+	"github.com/getsentry/sentry-go"
+
 	"pkg.world.dev/world-cli/cmd/world/root"
 )
 
 func main() {
+	// Sentry initialization
+	err := sentry.Init(sentry.ClientOptions{
+		Dsn:                "{input your DSN here}",
```
Comment (jerargus) [inline cmd/world/main.go:15]: This line breaks the tool (`sentry.Init: [Sentry] DsnParseError: invalid scheme`). Sentry should be disabled if a valid DNS isn't provided.

Comment (zulkhair) [reply cmd/world/main.go]: updated

**Review by jerargus — APPROVED** [review]

**Pull request merged**